zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirector.java

```diff
@@ -186,6 +187,44 @@ public ActorFuture<Void> resumeExporting() {
         });
   }
 
+  /**
+   * Disables an already configured exporter. No records will be exported to this exporter anymore.
+   *
+   * @param exporterId id of the exporter to disabled
+   * @return future which will be completed after the exporter is disabled.
+   */
+  public ActorFuture<Void> disableExporter(final String exporterId) {
+    if (actor.isClosed()) {
+      return CompletableActorFuture.completed(null);
+    }
+
+    return actor.call(() -> removeExporter(exporterId));
+  }
+
+  private void removeExporter(final String exporterId) {
+    final Optional<ExporterContainer> optionalContainer =
+        containers.stream().filter(c -> c.getId().equals(exporterId)).findFirst();
+
+    if (optionalContainer.isEmpty()) {
+      LOG.debug("Exporter '{}' is not found. It may be already removed.", exporterId);
+      return;
+    }
+
+    final var container = optionalContainer.get();
+    container.close();
+    containers.remove(container);
+    state.removeExporterState(exporterId);
+    // After removing this exporter, the exporter index has changed. Reset it so that we don't
+    // miss to export the record to any of the exporters whose index has changed.
+    recordExporter.resetExporterIndex();
+    LOG.debug("Exporter '{}' is removed.", exporterId);
+
+    if (containers.isEmpty()) {
+      LOG.info("No exporters are configured. Closing the exporter director '{}'.", name);
+      actor.close();
+    }
+  }
```

Comment:
:upside_down_face: Ultra extreme super soft suggestion, I'm just not a fan of using `Optional#isEmpty` (or `Optional#isPresent`) followed by `Optional#get`, I find the "functional" style more readable. But it's purely preference, not a fact or anything :sweat_smile: 

```suggestion
  private void removeExporter(final String exporterId) {
    containers.stream()
        .filter(c -> c.getId().equals(exporterId))
        .findFirst()
        .ifPresentOrElse(
            c -> removeExporter(exporterId, c),
            () -> LOG.debug("Exporter '{}' is not found. It may be already removed.", exporterId));
  }

  private void removeExporter(final String exporterId, final ExporterContainer container) {
    container.close();
    containers.remove(container);
    state.removeExporterState(exporterId);
    // After removing this exporter, the exporter index has changed. Reset it so that we don't
    // miss to export the record to any of the exporters whose index has changed.
    recordExporter.resetExporterIndex();
    LOG.debug("Exporter '{}' is removed.", exporterId);

    if (containers.isEmpty()) {
      LOG.info("No exporters are configured. Closing the exporter director '{}'.", name);
      actor.close();
    }
  }
```